### ISSUE TITLE
Color fix needed? 

### ISSUE THREAD

**Issue by hemanth** [altercation/solarized#220]
![color](http://i.imgur.com/8wsMl.png)

That's how it looks in my terminal, can read the text only after highlighting them! 

The color of the text is gray!

**Comment by hans-d** [comment]
+1

**Comment by isqua** [comment]
+1

**Comment by dotnil** [comment]
+1

But after 30min of reading these related issues I think it might be the time to ditch Solarized Darko for iTem2.

**Comment by dotnil** [comment]
https://github.com/chriskempson/base16-iterm2

**Comment by calfzhou** [comment]
+1

**Comment by nicolasmccurdy** [comment]
Does the base16 Solarized Dark theme have the same issues?

**Comment by cuviper** [comment]
It's fine with base16 because that makes use of extended colors.  You have
to do a little more work with base16-shell to get everything set up.

**Comment by nicolasmccurdy** [comment]
Cool, thanks for the quick answer.

**Comment by zevav** [comment]
@hydrozen 👍👍👍

**Comment by eugeneglova** [comment]
+1 contrast fix

**Comment by robmuh** [comment]
It is worth noting that Ubuntu 15.04 now officially includes Solarized in the standard selections for gnome-terminal color schemes. If you use Chalk or other themes that are not Solarized compatible you are choosing to be incompatible with a standard Ubuntu theme at this point. My experience has been only Chalk applications are subject to this issue. I have never encountered even an old app that uses terminal colors that is as broken as most Chalk applications.

**Comment by iamolivinius** [comment]
htop
Am 24.08.2015 4:40 nachm. schrieb "Rob Muhlestein" <notifications@github.com

**Comment by robmuh** [comment]
Thanks @iamolivinius haven't been using `htop` (mostly because I believe in teaching students tools that have more ubiquity, like `top` and `sar`, etc.) but this is indeed unfortunate. But I certainly won't be missing it.

**Comment by mrozbarry** [comment]
As a heads up to people using iTerm2 that have this issue and don't want to modify the theme, you can bump up the contrast about 10-20% in your terminal colour profile settings, and it made the invisible text readable without making a mess of the other colours.  It does get a little brighter looking, but it's reasonable.

**Comment by lvarayut** [comment]
+1 I'm having the same issue.

**Comment by guillaumevincent** [comment]
3 years after...
I fixed this using base16-solarized.dark.itermcolors found here : https://github.com/chriskempson/base16-iterm2 

It use extended color : https://github.com/altercation/solarized/issues/220#issuecomment-95610140

**Comment by thuandt** [comment]
I'm using Ubuntu 15.04, Guake Terminal and Terminator with Solarized Dark.
htop, goaccess and many another program has this issue.

I read whole thread but don't see how to fix this. And this thread was 3 years ago

**Comment by WouterSioen** [comment]
+1 with gulp

**Comment by eightyfive** [comment]
In Mac OS X Terminal Preferences > Profiles > Solarized Dark, just manually change the `Color Ansi (8)` to whatever best suits your eyes.

**Comment by unfulvio** [comment]
:+1:

**Comment by ishankhare07** [comment]
having similar issues with gnome-terminal on ubuntu 15.10

**Comment by NoAnyLove** [comment]
having the same issue with htop, any progress?

**Comment by desbma** [comment]
It's quite obvious this project is abandoned, last commit is from 2013.

**Comment by NoAnyLove** [comment]
Hi @desbma , it there any alternative project? I want to use it with MinTTY.

**Comment by Ahti** [comment]
@altercation do you have any opinion on this? I'm a huge fan of solarized, but with this issue still open, I can't really justify recommending it to friends atm.

If you don't have the time to work on this anymore, maybe try to find a new maintainer?

**Comment by dfournier** [comment]
This repo [hydrozen/solarized](https://github.com/hydrozen/solarized) doesn't exist anymore :cry: 
Any solution?

**Comment by nicolasmccurdy** [comment]
Lately, I have been enjoying [base16's default light theme](http://chriskempson.github.io/base16/). There is also a [base16 solarized theme](http://chriskempson.github.io/base16/#solarized), which may not have these issues.

**Comment by zevav** [comment]
@nicolasmccurdy thanks for the tip.  just spent 30 mins trying to get that to work. (similarly awful salmonesque results in vim)

<img width="643" alt="screen shot 2016-01-01 at 4 51 30 am" src="https://cloud.githubusercontent.com/assets/2838836/12070455/5c1f2a66-b043-11e5-9638-0fc73d798ad2.png">

**Comment by nicolasmccurdy** [comment]
Huh, so does their solarized theme seem to be a good alternative in general?

**Comment by zevav** [comment]
I can't get base16 to work in my terminal in the first place, so I can't say.  Looks great on the site!

**Comment by anarcat** [comment]
i have come to see this problem as un-solvable: the solarized color scheme is a hack, and it changes colors fundamentally in a way that breaks applications that expect certain colors to have certain contrast, by using weird background colors (e.g. iptraf). most of the time, i can switch color schemes in those to be simpler (e.g. B&W in iptraf), so i have mostly given up on this issue.

**Comment by ArcadeFever** [comment]
+1 for adjusting contrast in iterm2 to solve this

**Comment by pgarrison** [comment]
@dfournier You can still see the changes at #240. 

I solved this (at least among the tools I use) on urxvt by mapping color8 to base01 in my Xresources and tweaking the vim theme (pgarrison/vim-colors-solarized).

**Comment by metakermit** [comment]
If someone is still having trouble with this issue and since hydrozen's fork mentioned that fixes this is no longer available (the user is shown as @ghost now and the old https://github.com/hydrozen/solarized repo is gone), you might want to try the Solarized Dark Patched theme available in https://github.com/mbadolato/iTerm2-Color-Schemes . Just download [this file](https://raw.githubusercontent.com/mbadolato/iTerm2-Color-Schemes/master/schemes/Solarized%20Dark%20-%20Patched.itermcolors), enter your iTerm preferences -> Profiles -> Colors -> Load Presets... -> Import... and select & activate the theme. That's how I managed to solve the issue for myself.

**Comment by dfournier** [comment]
I moved to https://github.com/chriskempson/base16 with solorized theme.
Works perfectly!

**Comment by zeorin** [comment]
+1 for fixing this issue if possible. @cuviper's solution of switching which ANSI codes base02 base03 are mapped to seems the most reasonable.

I think Solarized works so well because of the careful colour choices, so any solution that changes the colours would not be the best, IMO. Switching the ANSI mappings would mean that any theme/colourscheme in a cli program that counts on the Solarized theme to be the terminal colours would have to be adjusted, sure.

But compared to the alternatives (choose a new colour that doesn't fit the thinking, change to a different colour scheme, change the whole contrast of the scheme, or just lose users over time) it's reasonable: it's a small change for "downstream" themes, and at least brblack will be visible.

Sure, one can argue that others shouldn't be putting important information in brblack, but it should at least be *visible*.

It's a non-backwards compatible-change, sure. These things happen in software projects. Let's handle it.

**Comment by icyflame** [comment]
@hemanth Can you please change the name of this issue? (Clearly, the fix is required. That would help in avoiding opening of new issues. I searched for the issue that I was facing and ended up opening #351 , until @zeorin informed me that the issue has already been reported.) Thanks!

And yeah, +1 for fixing this issue.

**Comment by anarcat** [comment]
for the record, i have mostly alleviated the problems here by switching from xterm to rxvt-unicode...

**Comment by zeorin** [comment]
Hi all,

I don’t think that, outside of forking this project ourselves, this issue or any other issues are likely to be fixed.

So that’s exactly what I’m proposing: let’s create a community-maintained fork!

I’ve created a new issue for the discussion of this idea: https://github.com/altercation/solarized/issues/355

Please do chime in.

**Comment by goyatparmod** [comment]
![image](https://cloud.githubusercontent.com/assets/13874876/17992059/52cb4bec-6b01-11e6-8f25-470a8d982140.png)

Left hand side is OSX terminal app and left hand side is iterm2. Font size and color scheme is same..
But color and font rendering is different. Any idea how i can fix OSX terminal color rendering.

**Comment by sheeit** [comment]
Here's a simple command to reproduce this:
if you have your terminal set up with the Solarized theme, just do

``` sh
printf 'Normal text. \033[1;30mInvisible text\033[0m.\n'

```

The text "Invisible text" will not be visible.


![solarized](https://user-images.githubusercontent.com/17221905/47673560-40f28800-dbb5-11e8-8e89-7f03627440dc.png)

**Comment by ivanmanolov90** [comment]
i love this theme but it has been 4 years since this issue was created, are there any plans to fix this?

**Comment by tyrsius** [comment]
@altercation Its been 4 years since you said you would take a look. Has there been any progress on this, or should we abandon solarized dark?

**Comment by icyflame** [comment]
hey @tyrsius, the possibility of this being fixed is remote. You should try using base16, I switched to that when I first encountered this issue and it's been pretty good. :smile:

**Comment by jacoscaz** [comment]
@ivanmanolov90's solutions seems to works perfectly for me - at least with the tools I use.

**Comment by bobf** [comment]
I "fixed" this by setting iTerm2's `black light` colour to `0387ad` (as suggested by @ivanmanolov90 ) and then using the following settings:
```
set background=dark
let g:solarized_termtrans = 1
colorscheme solarized
```
I know this won't work for everyone but if your setup is like mine then maybe it'll save you 15 minutes of frustration.

**Comment by icyflame** [comment]
I copied over the colors from gnome-terminal to iTerm2 and created a new color scheme. [here][1]. I tried using both [base16][2] and [base256][3] for iTerm2, but the first one simply didn't show some text and the second one had a weird problem where the difference between folders and files was very _small_ (they were both white and the difference in font weight was not noticeable at all)

I have been using this for a couple days now, and it's not shown up any problems. I will report here if it does.

P.S the copying over process was pretty simple and you should absolutely do it if you have a preferred setup on a computer that works fine and want to emulate the setup elsewhere. (Not sure if this was already discussed here, I was unable to find it anywhere) :thinking: 

[1]: https://github.com/icyflame/dotfiles/blob/fa9ed806dc3d3bc98056f10a99a306961967ea01/color-schemes/iterm2/iterm2-solarized-manual.itermcolors
[2]: https://github.com/martinlindhe/base16-iterm2/blob/master/itermcolors/base16-solarized-dark.itermcolors
[3]: https://github.com/martinlindhe/base16-iterm2/blob/master/itermcolors/base16-solarized-dark-256.itermcolors

**Comment by laggardkernel** [comment]
> I copied over the colors from gnome-terminal to iTerm2 and created a new color scheme. [here](https://github.com/icyflame/dotfiles/blob/fa9ed806dc3d3bc98056f10a99a306961967ea01/color-schemes/iterm2/iterm2-solarized-manual.itermcolors). 

I see. You're using the normal colors as the bright ones. I'm going to do the same. At least, the bright colors are not grey anymore.

**Comment by timber-they** [comment]
Mocha is pretty much the only tool where this issue still occurs
